packages/components/src/mobile/bottom-sheet/index.native.js

```diff
@@ -382,7 +382,9 @@ class BottomSheet extends Component {
 				onSwipe={ this.onCloseBottomSheet }
 				onDismiss={ Platform.OS === 'ios' ? onDismiss : undefined }
 				onModalHide={
-					Platform.OS === 'android' ? onDismiss : undefined
+					this.props.onModalHide || Platform.OS === 'android'
```

Comment:
This is kinda confusing. You could use `rest.onModalHide` since props are destructured to some props and the `rest`.
However, [here](https://github.com/WordPress/gutenberg/pull/27604/files#diff-7c783d6dab99fab7810467451bf92877850e8c2f4808af9dd836ca828fdd9821R399) we pass the `rest` to the Modal, and the `onModalHide` prop will be overwritten. I'm not sure if adding this check makes sense.